### PR TITLE
Creating game servers samples.

### PR DESCRIPTION
**Pull request by dzlier-gcp**

### CODE REVIEW

**Comment by dzlier-gcp** [comment]
@kurtisvg kvg bot assigned me to my own PR lol.

File: gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      RetryingFuture<OperationSnapshot> poll = client.createAllocationPolicyAsync(
+          CreateAllocationPolicyRequest
+              .newBuilder()
+              .setParent(parent)
+              .setAllocationPolicyId(policyId)
+              .setAllocationPolicy(policy)
+              .build()).getPollingFuture();
+
+      if (poll.get(1, TimeUnit.MINUTES).isDone()) {
+        System.out.println("Allocation Policy created: " + policy.getName());
```
[the remark this answers is not in the document]
Comment (dzlier-gcp) [reply gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java]: Yeah, the create call has a response of a protobuf Any, which I can unpack into thew new  policy.

[the remark this answers is not in the document]
Comment (dzlier-gcp) [reply gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java]: isDone just means completed, so I can add a check on the status code as well.

File: gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          .setPriority(Int32Value.newBuilder().setValue(1))
+          .build();
+
+      RetryingFuture<OperationSnapshot> poll = client.createAllocationPolicyAsync(
+          CreateAllocationPolicyRequest
+              .newBuilder()
+              .setParent(parent)
+              .setAllocationPolicyId(policyId)
+              .setAllocationPolicy(policy)
+              .build()).getPollingFuture();
```
[the remark this answers is not in the document]
Comment (dzlier-gcp) [reply gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java]: Putting a timeout on the `get` call means it throws a TimeoutException if the call doesn't complete. I believe this is the actual future for the call though, not something that polls for updates itself, and the polling is part of retrying if the request fails. But I could be wrong.

File: gameservices/v1alpha/pom.xml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  <name>gameservices</name>
+
+  <!--
+    The parent pom defines common style checks and testing strategies for our samples.
+    Removing or replacing it should not affect the execution of the samples in anyway.
+  -->
+  <parent>
+    <groupId>com.google.cloud.samples</groupId>
+    <artifactId>shared-configuration</artifactId>
+    <version>1.0.10</version>
```
[the remark this answers is not in the document]
Comment (kurtisvg) [reply gameservices/v1alpha/pom.xml]: Was this missed?

File: gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateAllocationPolicy {
+  public static void createAllocationPolicy(String projectId, String policyId) {
+    // String projectId = "your-project-id";
+    // String policyId = "your-policy-id";
+    try (AllocationPoliciesServiceClient client = AllocationPoliciesServiceClient.create()) {
```
Comment (kurtisvg) [inline gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java]: Please add a clarification comment letting the user know how to reuse clients (see https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#client-initialization)

File: gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      String parent = String.format("projects/%s/locations/global", projectId);
+      String policyName = String.format("%s/allocationPolicies/%s", parent, policyId);
```
Comment (kurtisvg) [inline gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java]: FYI - it looks like there is a helper method for this: [`AllocationPolicyName.of()`](https://googleapis.dev/java/google-api-grpc/latest/com/google/cloud/gaming/v1alpha/AllocationPolicyName.html#of-java.lang.String-java.lang.String-java.lang.String-)

Might be easier to read/use - up to you.

File: gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+      AllocationPolicy policy = AllocationPolicy
+          .newBuilder()
+          .setName(policyName)
+          .setPriority(Int32Value.newBuilder().setValue(1))
+          .build();
+
+      CreateAllocationPolicyRequest request = CreateAllocationPolicyRequest
```
Comment (kurtisvg) [inline gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java]: FYI - it looks like there is a nice little [helper that avoids building a request. ](https://googleapis.dev/java/google-cloud-clients/latest/com/google/cloud/gaming/v1alpha/AllocationPoliciesServiceClient.html#createAllocationPolicyAsync-java.lang.String-java.lang.String-com.google.cloud.gaming.v1alpha.AllocationPolicy-)

Again, up to you.

File: gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+          .newBuilder()
+          .setParent(parent)
+          .setAllocationPolicyId(policyId)
+          .setAllocationPolicy(policy)
+          .build();
+
+      RetryingFuture<OperationSnapshot> poll = client.createAllocationPolicyAsync(request)
```
Comment (kurtisvg) [inline gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java]: Ok so digging into this a bit, it looks like `createAllocationPolicyAsync` already returns an `OperationFuture`. The `getPollingFuture` actually returns a `[RetryingFuture](http://googleapis.github.io/gax-java/latest/apidocs/com/google/api/gax/retrying/RetryingFuture.html)` which is marked as "internal/advanced use only." So I'm still not sure what the PollingFuture does, but I'm pretty sure it's the wrong one. 

I think you probably want something like the following: 
```java
// Send the request to the server to start creating the AllocationPolicy
OperationFuture<AllocationPolicy> futureResponse = client.client.createAllocationPolicyAsync(request);
try {
    // Wait until the AllocationPolicy is created successfully
    AllocationPolicy response = futureResponse.get(1, TimeUnit.MINUTES);
    System.out.println("Allocation Policy created successfully: " + response.getName());
} catch (InterruptedException | TimeoutException e) {
    System.out.println("Unable to receive response from server: " + e);
}
```
I can't quite tell if the `create` api call is actually being performed asynchronously or if it is synchronous and the future represents a constant polling until complete. It is hard to tell from the documentation.

File: gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java (hunk 4 above)
Comment (dzlier-gcp) [reply gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java]: Strange, I do remember updating it to 1.0.11, but I guess it fell off. /shrug

File: gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java (hunk 5 above)
Comment (dzlier-gcp) [reply gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java]: I think I prefer the consistency with the other samples of formatting the parent name and then the resource name using the parent name.

File: gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java (hunk 7 above)
Comment (dzlier-gcp) [reply gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/allocationpolicies/CreateAllocationPolicy.java]: All right, I changed these out for just getting the response from the call directly.

**Comment by dzlier-gcp** [comment]
Ping - would like to get this canonical submitted soon. Thanks!

**Comment by kurtisvg** [comment]
> Ping - would like to get this canonical submitted soon. Thanks!

Sorry, I'm never quite sure when PRs are ready for review and the notifications get caught in my filters. You can use the "re-request review" button to make sure it shows up promptly.

File: gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/clusters/UpdateCluster.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+      GameServerCluster cluster = GameServerCluster
+          .newBuilder()
+          .setName(clusterName)
+          .putLabels("key", "value")
+          .build();
+
+      FieldMask fieldMask = FieldMask.newBuilder().addPaths("labels").build();
+      OperationFuture<GameServerCluster, Empty> call = client.updateGameServerClusterAsync(
+          cluster, fieldMask);
+
```
Comment (kurtisvg) [inline gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/clusters/UpdateCluster.java]: nit: remove extra empty line?

Comment (dzlier-gcp) [reply gameservices/v1alpha/src/main/java/com/google/cloud/gameservices/samples/clusters/UpdateCluster.java]: Will do with the re-request button. Thanks for your attention on this. I'll take care of the line.

**Comment by dzlier-gcp** [comment]
@dpebot Merge when green.

**Pull request merged**